File: testdata/nodejs/secure.js
// Properly configured TLS - should not trigger critical/high findings.

const tls = require("tls");
const https = require("https");

// Certificate verification is enabled by default.
// Always use modern TLS versions.

const secureOptions = {
  rejectUnauthorized: true,
  minVersion: "TLSv1.3",
};

function logStatus() {
  console.log("TLS verification is active");
  console.log("Using secure protocol version");
}

const server = tls.createServer(secureOptions);

File: testdata/go/secure.go
package secure

import (
	"crypto/tls"
	"log"
)

// Properly configured TLS - none of these should trigger critical/high findings.

func secureTLS() *tls.Config {
	cfg := &tls.Config{
		MinVersion: tls.VersionTLS12,
	}
	return cfg
}

func strictTLS() *tls.Config {
	return &tls.Config{
		InsecureSkipVerify: false,
		MinVersion:         tls.VersionTLS12,
	}
}

func warnUser() {
	log.Println("Certificate verification is enabled")
	log.Println("Using modern TLS version")
}
